lib/ae_mdw_web/websocket/broadcaster.ex

```diff
@@ -24,72 +29,81 @@ defmodule AeMdwWeb.Websocket.Broadcaster do
-  def broadcast_micro_block(block, source) do
-    GenServer.cast(__MODULE__, {:broadcast_micro_block, block, source})
+  @spec broadcast_micro_block(Db.micro_block(), version(), source()) :: :ok
+  def broadcast_micro_block(block, version, source) do
+    GenServer.cast(__MODULE__, {:broadcast_micro_block, version, block, source})
   end
 
-  @spec broadcast_txs(tuple(), :node | :mdw) :: :ok
-  def broadcast_txs(block, source) do
-    GenServer.cast(__MODULE__, {:broadcast_txs, block, source})
+  @spec broadcast_txs(Db.micro_block(), version(), source()) :: :ok
+  def broadcast_txs(block, version, source) do
+    GenServer.cast(__MODULE__, {:broadcast_txs, version, block, source})
   end
 
   @impl GenServer
-  def handle_cast({:broadcast_key_block, block, source}, state) do
+  def handle_cast({:broadcast_key_block, version, block, source}, state) do
     {:ok, hash} = block |> :aec_blocks.to_header() |> :aec_headers.hash_header()
 
-    if not already_processed?({:key, hash, source}) do
-      do_broadcast_key_block(block, source)
-      set_processed({:key, hash, source})
+    if not already_processed?({:key, version, hash, source}) do
+      do_broadcast_key_block(block, version, source)
+      set_processed({:key, version, hash, source})
     end
 
     {:noreply, state}
   end
 
   @impl GenServer
-  def handle_cast({:broadcast_micro_block, block, source}, state) do
+  def handle_cast({:broadcast_micro_block, version, block, source}, state) do
     {:ok, hash} = block |> :aec_blocks.to_header() |> :aec_headers.hash_header()
 
-    if not already_processed?({:micro, hash, source}) do
-      do_broadcast_micro_block(block, source)
-      set_processed({:micro, hash, source})
+    if not already_processed?({:micro, version, hash, source}) do
+      do_broadcast_micro_block(block, version, source)
+      set_processed({:micro, version, hash, source})
     end
 
     {:noreply, state}
   end
 
   @impl GenServer
-  def handle_cast({:broadcast_txs, block, source}, state) do
+  def handle_cast({:broadcast_txs, version, block, source}, state) do
     {:ok, hash} = block |> :aec_blocks.to_header() |> :aec_headers.hash_header()
 
-    if not already_processed?({:txs, hash, source}) do
-      do_broadcast_txs(block, source)
-      set_processed({:txs, hash, source})
+    if not already_processed?({:txs, version, hash, source}) do
+      do_broadcast_txs(block, version, source)
+      set_processed({:txs, version, hash, source})
     end
 
     {:noreply, state}
   end
 
   @impl GenServer
-  def handle_info({:broadcast_objects, header, tx, source}, state) do
-    ser_tx = :aetx_sign.serialize_for_client(header, tx)
-
-    tx
-    |> get_ids_from_tx()
-    |> Enum.each(&broadcast(&1, encode_message(ser_tx, "Object", source)))
-
+  def handle_info(_msg, state) do
     {:noreply, state}
   end
 
   #
   # Private functions
   #
-  defp do_broadcast_key_block(block, source) do
+  defp do_broadcast_key_block(block, :v2, :mdw) do
```

Comment:
would invert using `:v1` in the parameter with specific implementation and leave default (without pattern matching) for `:v2` to allow `:v3` continue from `:v2`

Reply:
The problem with this is that there's 4 possible outcomes (`node/v1`, `mdw/v1`, `node/v2` and `mdw/v2`) and the representation only differs with `mdw/v2` which is why it has a special clause

Reply:
It is in regards to a possible and likely `:v3` where the key block might be the same as the `:v2`.

If the `:v1` is the pattern matching, `:v3` would go to the default as `:v2`